Tests/Unit/Entity/CalendarTest.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace CanalTP\MttBundle\Tests\Unit\Entity;
+
+use CanalTP\MttBundle\Entity\Calendar;
+use Symfony\Component\Validator\Validation;
+
+class CalendarTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * Path to the validation file
+     *
+     * @var string
+     */
+    private static $validatorPath;
+
+    /**
+     * @var Calendar
+     */
+    private $calendar;
+
+    /**
+     * Initializes validation file path
+     */
+    public static function setUpBeforeClass()
+    {
+        parent::setUpBeforeClass();
+        self::$validatorPath = realpath(__DIR__.'/../../../').'/Resources/config/validation.yml';
+    }
+
+    /**
+     * Initializes $calendar and $validator
+     */
+    protected function setUp()
+    {
+        if (version_compare(PHP_VERSION, '7.0.0') === 1 ) {
+            $this->markTestSkipped();
+        }
+
+        $this->calendar = new Calendar();
+        $this->calendar
+            ->setTitle('CanalTP')
+            ->setStartDate(new \DateTime('2016-07-08'))
+            ->setEndDate(new \DateTime('2016-08-08'));
+
+        $this->validator = Validation::createValidatorBuilder()
+            ->addYamlMapping(self::$validatorPath)
+            ->getValidator();
+    }
+
+    /**
+     * Tests weekly pattern validity for calendar entity
+     *
+     * @dataProvider getPatterns
+     */
+    public function testWeeklyPatternValidity($pattern, $errorsNumber)
```

Comment:
$errorsNumber can become $valid because there is never more than one (same for testStartDateAndEndDateValidity and testTitleValidity)